libethereum/Client.cpp

```diff
@@ -119,9 +119,23 @@ Client::Client( ChainParams const& _params, int _networkID,
 }
 
 Client::~Client() {
+    stopWorking();
+}
+
+void Client::stopWorking() {
```

Comment:
I just realized that first thing that we need to do in Client's stopWorking - is to call m_skaleHost->stopWorking(). This is very important - because it will ensure that we are not in the middle of block importing! (to archieve this - SkaleHost waits for m_consensusWorkingMutex to become available). Also, important that join()'ing of m_snapshotHashComputing was done after this too.

Reply:
I trust you here.... resolved... nice catch!